src/Components/CreateImageWizard/formComponents/ActivationKeys.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormGroup, Spinner, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
+import api from '../../../api';
+
+const ActivationKeys = ({ label, isRequired }) => {
+    const { change } = useFormApi();
+    const [ activationKeys, setActivationKeys ] = useState([]);
+    const [ isOpen, setIsOpen ] = useState(false);
+    const [ isLoading, setIsLoading ] = useState(false);
+    const [ activationKeySelected, selectActivationKey ] = useState();
+
+    useEffect(() => {
+        setIsLoading(true);
+        const data = api.getActivationKeys();
+        data.then(keys => {
+            setActivationKeys(keys);
+            setIsLoading(false);
+        });
+    }, []);
+
+    const setActivationKey = (_, selection) => {
+        selectActivationKey(selection);
+        setIsOpen(false);
+        change('subscription-activation-key', selection);
+    };
+
+    const handleClear = () => {
+        selectActivationKey();
+        change('subscription-activation-key', undefined);
+    };
+
+    return (
+        <FormGroup isRequired={ isRequired } label={ label } data-testid='subscription-activation-key'>
+            <Select
+                variant={ SelectVariant.typeahead }
+                onToggle={ () => setIsOpen(!isOpen) }
+                onSelect={ setActivationKey }
+                onClear={ handleClear }
+                selections={ activationKeySelected }
+                isOpen={ isOpen }
+                placeholderText="Select activation key"
+                typeAheadAriaLabel="Select activation key">
+                {isLoading &&
+                    <SelectOption isNoResultsOption={ true } data-testid='activation-keys-loading'>
+                        <Spinner isSVG size="lg" />
+                    </SelectOption>
+                }
+                {activationKeys.map((key, index) => (
+                    <SelectOption
+                        key={ index }
+                        value={ key.id }
```

Comment:
I'd actually make the key.name the value and the description the id. Not sure the id is even exposed on access.

Reply:
I was unsure about that. So if a user was to type you think they would type in the descriptive value?

Reply:
Agreed. I don't think the id is ever used except internally by RH, right? I.e., the thing we embed in our images is the name, not the ID, so the customers would not see them? (not sure I understand how that makes sense, but that's another matter)

Reply:
Okay cool. Changed.

Reply:
I switched them but should we remove the id completely?

Reply:
Hm... Maybe? If it is not shown anywhere else, I would. Are the names forced to be unique?

Reply:
Only displaying the names now.

Reply:
Per this mock: https://marvelapp.com/prototype/6g7dja9/screen/76602546